### PR TITLE
Travis/AppVeyor: Explicitly specify all build options

### PR DESCRIPTION
**Pull request by uklotzde**
...for validation and documentation purposes. Otherwise we wouldn't notice if some feature flags are deactivated implicitly due to errors in the setup of the build environment or the golden environment.

Additional changes:
- Follow-up of closed PR #2890: Restrict installation of homebrew build environment to the legacy macOS SCons build (to be removed). The CMake build already uses the golden environment from our build server.
- Use -DWARNINGS_FATAL for Ubuntu2004 build on AppVeyor

### CODE REVIEW

**Comment by uklotzde** [comment]
Finally ready. Specifying all options explicitly provided some surprising insights **what** features are actually available and enabled on each platform and build system.

File: appveyor.yml [hunk 1]
```diff
@@ -107,12 +107,22 @@ for:
     - cd cmake_build
     - cmake
       -L
+      -DWARNINGS_FATAL=ON
+      -DDEBUG_ASSERTIONS_FATAL=ON
```
Comment (Holzhaus) [inline appveyor.yml:111]: Note that we will need to disable this when the we start to use CI for deployment. Maybe we enable conditionally for PRs but not for master builds/deployments.

**Comment by uklotzde** [comment]
@Holzhaus We do not explicitly set CMAKE_BUILD_TYPE, only for the Windows build on AppVeyor. How about setting this explicitly and where (env, flags, ...)?

File: appveyor.yml (hunk 1 above)
Comment (uklotzde) [reply appveyor.yml:112]: Added TODO comments

**Comment by Holzhaus** [comment]
> @Holzhaus We do not explicitly set CMAKE_BUILD_TYPE, only for the Windows build on AppVeyor. How about setting this explicitly and where (env, flags, ...)?

Maybe we should adapt CMakeLists.txt to set it when it hasn't been specified? https://blog.kitware.com/cmake-and-the-default-build-type/

**Comment by uklotzde** [comment]
> > @Holzhaus We do not explicitly set CMAKE_BUILD_TYPE, only for the Windows build on AppVeyor. How about setting this explicitly and where (env, flags, ...)?
> 
> Maybe we should adapt CMakeLists.txt to set it when it hasn't been specified? https://blog.kitware.com/cmake-and-the-default-build-type/

Nevertheless we should explicitly define it for the CI builds.

**Comment by Holzhaus** [comment]
For now let's use `-DCMAKE_BUILD_TYPE=Debug` and put it into the common flags env var.

**Comment by Holzhaus** [comment]
Okay, release it is.

**Review by Holzhaus — APPROVED** [review]
LGTM. Waiting for CI.

**Comment by uklotzde** [comment]
Explicitly disabling unused features introduces a lot of noise. I have decided to remove them. Let's see how this works.

**Comment by uklotzde** [comment]
@Holzhaus Ping

**Pull request merged**